### PR TITLE
Run stellar mart DAG at the beginning of day

### PR DESCRIPTION
**Pull request by amishas157**
<details>
  <summary>PR Checklist</summary>

### PR Structure

- [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
- [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
      otherwise).
- [ ] This PR's title starts with the jira ticket associated with the PR.

### Thoroughness

- [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
- [ ] I've updated the README with the added features, breaking changes, new instructions on how to use the repository.

</details>

### What

This PR:
- Updates the schedule of stellar mart dag so as to refresh data earlier
- Updates the schedule of recency tests to receive any alerts early in work day
- Separate MGI task to another DAG based on time when mgi files are received

### Why

To refresh data as early as possible

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: dags/dbt_recency_tests_dag.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
     default_args=get_default_dag_args(),
     start_date=datetime(2024, 9, 24, 0, 0),
     description="This DAG runs recency tests at a daily cadence",
-    schedule="0 20 * * *",  # Every day at 8:00 PM UTC / 3:00 PM CST
+    schedule="0 15 * * *",  # Every day at 3:00 PM UTC / 9:00 AM CST
```
Comment (sydneynotthecity) [inline dags/dbt_recency_tests_dag.py:21]: How did you land on 15:00 UTC? Do you think we could run even earlier? (Like 2:00 or 3:00)

File: dags/dbt_stellar_marts_mgi_dag.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    default_args=get_default_dag_args(),
+    start_date=datetime(2024, 10, 1, 0, 0),
+    description="This DAG runs dbt models at a daily cadence",
+    schedule_interval="0 17 * * *",  # Runs at 05:00 PM UTC / 11:00 AM CST as MGI files are received around ~10:00 AM CST
+    user_defined_filters={
+        "container_resources": lambda s: k8s.V1ResourceRequirements(requests=s),
+    },
+    max_active_runs=3,
+    catchup=True,
+    tags=["dbt-stellar-marts"],
```
Comment (sydneynotthecity) [inline dags/dbt_stellar_marts_mgi_dag.py:26]: Do we need to have different tags for a different DAG? What does this impact by keeping the same tags?

File: dags/dbt_stellar_marts_dag.py [hunk 3]
```diff
@@ -58,9 +57,6 @@
 # DAG task graph
 wait_on_dbt_enriched_base_tables >> ohlc_task >> liquidity_pool_trade_volume_task
 
-wait_on_dbt_enriched_base_tables >> mgi_task
-wait_on_partner_pipeline_dag >> mgi_task
```
Comment (sydneynotthecity) [inline dags/dbt_stellar_marts_dag.py:62]: I think we can also delete the wait on partner pipeline dag task

**Review by sydneynotthecity — APPROVED** [review]
Couple nits, nothing blocking the merge

File: dags/dbt_recency_tests_dag.py (hunk 1 above)
Comment (amishas157) [reply dags/dbt_recency_tests_dag.py:21]: Selected 15:00 UTC to have the elementary run at 9 CST. Mostly for convenience to get alerts as we start workday. But can definitely be changed to 2:00/3:00

File: dags/dbt_stellar_marts_mgi_dag.py (hunk 2 above)
Comment (amishas157) [reply dags/dbt_stellar_marts_mgi_dag.py:26]: It just helps to narrow down list of dags in airflow UI. https://airflow.apache.org/docs/apache-airflow/stable/howto/add-dag-tags.html

File: dags/dbt_recency_tests_dag.py (hunk 1 above)
Comment (sydneynotthecity) [reply dags/dbt_recency_tests_dag.py]: That's fine. we can leave the schedule as is and if we run into issues move it up

**Pull request merged**